aws/s3/lock.go

```diff
@@ -4,12 +4,36 @@ import (
 	"fmt"
 
 	"github.com/coinbase/step/aws"
+	"github.com/coinbase/step/errors"
 )
 
 type Lock struct {
 	UUID string `json:"uuid,omitempty"`
 }
 
+type UserLock struct {
+	User       string `json:"user,omitempty"`
+	LockReason string `json:"lock_reason", omitempty"`
+}
+
+func CheckUserLock(s3c aws.S3API, bucket *string, lock_path *string) error {
+	var userLock UserLock
+	err := GetStruct(s3c, bucket, lock_path, &userLock)
+	if err != nil {
+		switch err.(type) {
+		case *NotFoundError:
+			// good we want this
+		default:
+			return err // All other errors return
+		}
+	}
+	if userLock.User != "" {
```

Comment:
I think this should not be an if, because it is locked if the lock-file exists. If the user is there or not.